astropy/io/ascii/qdp.py

```diff
@@ -136,9 +138,8 @@ def _get_type_from_list_of_lines(lines, delimiter=None):
 def _get_lines_from_file(qdp_file):
     if "\n" in qdp_file:
         lines = qdp_file.split("\n")
-    elif isinstance(qdp_file, str):
-        with open(qdp_file) as fobj:
-            lines = [line.strip() for line in fobj]
+    elif isinstance(qdp_file, (str, os.PathLike)):
+        lines = [line.strip() for line in Path(qdp_file).read_text().splitlines()]
```

Comment:
This new code ends up making two additional copies of the entire file in memory, one the full `read_text()` output and second the `splitlines()` output. In general `splitlines()` is not a drop-in replacement for iterating over `open()` since `splitlines()` removes the line endings. In this case it doesn't matter but best to keep this `Path` refactoring as close to the original code as possible.

So basically the only change here should be `isinstance(qdp_file, (str, os.PathLike)):`.

Reply:
> In this case it doesn't matter but best to keep this Path refactoring as close to the original code as possible.

100% agreed. Any behavioral difference with the original code that I don't advertise myself in a comment is a mistake from me, so thank you for your attention to details !